docs/Datagrid.md

```diff
@@ -0,0 +1,737 @@
+
+```jsx
+// in src/PostList.js
+import { Datagrid, DatagridBody, List, TextField } from 'react-admin';
+import TableCell from '@material-ui/core/TableCell';
+import TableRow from '@material-ui/core/TableRow';
+import Checkbox from '@material-ui/core/Checkbox';
+
+const MyDatagridRow = ({ record, resource, id, onToggleItem, children, selected, selectable, basePath }) => (
+    <TableRow key={id}>
+        {/* first column: selection checkbox */}
+        <TableCell padding="none">
+            <Checkbox
+                disabled={selectable}
+                checked={selected}
+                onClick={event => onToggleItem(id, event)}
+            />
+        </TableCell>
+        {/* data columns based on children */}
+        {React.Children.map(children, field => (
+            <TableCell key={`${id}-${field.props.source}`}>
+                {field}
+            </TableCell>
+        ))}
+    </TableRow>
+);
+
+const MyDatagridBody = props => <DatagridBody {...props} row={<MyDatagridRow />} />;
+const MyDatagrid = props => <Datagrid {...props} body={<MyDatagridBody />} />;
+
+const PostList = () => (
+    <List>
+        <MyDatagrid>
+            <TextField source="title" />
+            ...
+        </MyDatagrid>
+    </List>
+)
+
+export default PostList;
+```
+
+## `bulkActionButtons`
+
+![Bulk Action Buttons](./img/bulk-actions-toolbar.gif)
+
+Bulk action buttons are buttons that affect several records at once, like mass deletion for instance. In the `<Datagrid>` component, the bulk actions toolbar appears when a user ticks the checkboxes in the first column of the table. The user can then choose a button from the bulk actions toolbar. By default, all datagrids have a single bulk action button, the bulk delete button. You can add other bulk action buttons by passing a custom element as the `bulkActionButtons` prop of the `<List>` component:
+
+```jsx
+import { Button } from '@mui/material';
+import { BulkDeleteButton } from 'react-admin';
+
+import ResetViewsButton from './ResetViewsButton';
+
+const PostBulkActionButtons = () => (
+    <>
+        <ResetViewsButton label="Reset Views" />
+        {/* default bulk delete action */}
+        <BulkDeleteButton />
+    </>
+);
+
+export const PostList = () => (
+    <List>
+        <Datagrid bulkActionButtons={<PostBulkActionButtons />}>
+            ...
+        </Datagrid>
+    </List>
+);
+```
+
+**Tip**: React-admin provides three components that you can use in `bulkActionButtons`: `<BulkDeleteButton>`, `<BulkUpdateButton>`, and `<BulkExportButton>`.
+
+**Tip**: You can also disable bulk actions altogether by passing `false` to the `bulkActionButtons` prop. In this case, the checkboxes column doesn't show up.
+
+Bulk action button components can use the [`useListContext`](./uselistcontext.md) hook to get the elements they need to perform their job:
```

Comment:
`useListContext.md` yet to be created? => \[`useListContext`\]\(./useListContext.md\)